clap_complete/tests/testsuite/engine.rs

```diff
@@ -981,6 +981,24 @@ a_pos,c_pos"
     );
 }
 
+#[test]
+fn suggest_external_subcommand() {
+    let mut cmd = Command::new("dynamic")
+        .arg(clap::Arg::new("positional").value_parser(["pos1", "pos2", "pos3"]));
```

Comment:
This is not a subcommand and adding subcommand candidates when no subcommands are taken is strange.  I wonder if  we should gate the reading of `SubcommandCandidates` based on whether external subcommands are allowed.